bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/handler/VehicleHandler.java

```diff
@@ -291,7 +291,8 @@ public State getValue(String channelId, Status status, VehiclePositionWrapper po
             case FUEL_AMOUNT:
                 return new QuantityType<Volume>(status.fuelAmount, SmartHomeUnits.LITRE);
             case FUEL_LEVEL:
-                return new QuantityType<>(status.fuelAmountLevel, SmartHomeUnits.PERCENT);
+                return status.fuelAmountLevel != -1 ? new QuantityType<>(status.fuelAmountLevel, SmartHomeUnits.PERCENT)
+                        : UnDefType.UNDEF;
             case FUEL_CONSUMPTION:
                 return new DecimalType(status.averageFuelConsumption.floatValue() / 10);
```

Comment:
Double values should not be converted to float, better use `doubleValue` or maybe better just use a primitive field `double`.